src/views/htmlcontent/src/js/components/messagescontextmenu.component.ts

```diff
@@ -0,0 +1,86 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+import { Component, Output, EventEmitter, Inject, forwardRef, OnInit } from '@angular/core';
+
+import {ShortcutService} from './../services/shortcuts.service';
+
+import * as Constants from './../constants';
+import * as Utils from './../utils';
+
+
+import { IRange } from './../interfaces';
+
+/**
+ * The component that acts as the contextMenu for slick grid
+ */
+
+const template = `
+<ul class="contextMenu" style="position:absolute" [class.hidden]="!visible" [style.top]="position.y" [style.left]="position.x">
+    <li id="copy" (click)="handleContextActionClick('copySelection')" [class.disabled]="isDisabled"> {{Constants.copyLabel}}
+        <span style="float: right; color: lightgrey; padding-left: 10px">{{keys['event.copySelection']}}</span>
+    </li>
+</ul>
+`;
+
+@Component({
+    selector: 'msg-context-menu',
+    providers: [ShortcutService],
+    template: template
+})
+
+export class MessagesContextMenu implements OnInit {
+    // tslint:disable-next-line:no-unused-variable
+    private Utils = Utils;
+    // tslint:disable-next-line:no-unused-variable
+    private Constants = Constants;
+
+    @Output() clickEvent: EventEmitter<{type: string, selectedRange: IRange }>
+        = new EventEmitter<{type: string, selectedRange: IRange}>();
+    private selectedRange: IRange;
+    private isDisabled: boolean;
+    private position: {x: number, y: number} = {x: 0, y: 0};
+    private visible: boolean = false;
+    private keys = {
+        'event.copySelection': ''
+    };
+
+
+    constructor(@Inject(forwardRef(() => ShortcutService)) private shortcuts: ShortcutService) {
+        const self = this;
+        for (let key in this.keys) {
+            if (this.keys.hasOwnProperty(key)) {
+                this.shortcuts.stringCodeFor(key).then((result) => {
+                    self.keys[key] = result;
+                });
+            }
+        }
+    }
+
+    ngOnInit(): void {
+        const self = this;
+        $(document).on('click', () => {
+            self.hide();
+        });
+    }
+
+    show(x: number, y: number, selectedRange: IRange): void {
+        this.selectedRange = selectedRange;
+        let selectedText = (selectedRange && selectedRange.toString) ? selectedRange.toString() : '';
+        this.isDisabled = selectedText.length === 0;
```

Comment:
This context menu poses an interesting scenario if text isn't selected because `Copy` is the only option. It looks like different applications handle context menu copying differently - some don't list `Copy` if text is not selected, and some do what you did and grey it out. I think the current behavior is probably best, but I just want to make sure you thought of the possibility of disabling the context menu when there is no text selection.